cpp/src/Slice/Parser.cpp

```diff
@@ -791,7 +800,14 @@ Slice::Contained::parseDocComment(function<string(string, string)> linkFormatter
         }
         else if (parseCommentLine(l, throwsTag, true, name, lineText))
         {
-            if (!lineText.empty())
+            if (!isOperation)
+            {
+                // If '@throws' was put on anything other than an operation, ignore it and issue a warning.
+                string msg = "the '" + throwsTag + "' tag is only valid on operations";
+                unit()->warning(file(), line(), InvalidComment, msg);
+                state = StateUnknown;
+            }
+            else if (!lineText.empty())
```

Comment:
Should we emit a warning for empty throws?

Reply:
The idea behind this is for users who format their comments like:
```
@throw AlreadyActivatedException
This exception is thrown if the thing is already activated
```
Just because the rest of the line is empty, doesn't mean that the section as a whole is empty.
It could just be on the next line down.

Reply:
Nope, but I see what you were thinking now. `lineText` only includes the text that comes _after_ the name of the exception, parameter, type, etc.

The function we call is
`bool parseCommentLine(const string& l, const string& tag, bool namedTag, string& name, string& doc)`
and it has 2 'out parameters' `name` and `doc`. In this case `name` will be the name of the exception, and `doc` (which is `lineText` here) is anything that comes after that name.